src/NewRemoting/RemoteLoaderWindowsClient.cs

```diff
@@ -185,6 +185,16 @@ public void Connect(CancellationToken externalToken, ILogger clientConnectionLog
 				{
 					Logger.LogError(x, $"Unable to connect to remote server. Attempt {retries + 1}: {x.Message}");
 					lastError = x;
+
+					if (process is { ExitCode: -1 })

```

Comment:
Wouldn't this throw an exception if the process didn't terminate yet?

Reply:
true